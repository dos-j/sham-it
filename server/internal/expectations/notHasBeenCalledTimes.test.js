const notHasBeenCalledTimes = require("./notHasBeenCalledTimes");

describe("unit: notHasBeenCalledTimes", () => {
  let handler;
  let requestStore;

  beforeEach(() => {
    requestStore = [];
    handler = notHasBeenCalledTimes(requestStore);
  });

  test("it should return 204 No Content if the times given matches the number of requests in the requestStore", () => {
    requestStore.push({
      request: {
        method: "GET",
        pathname: "/test",
        query: { a: 1234 },
        headers: { accept: "application/json" }
      }
    });
    requestStore.push({
      request: {
        method: "GET",
        pathname: "/test",
        query: { a: 1234 },
        headers: { accept: "application/json" }
      }
    });
    requestStore.push({
      request: {
        method: "GET",
        pathname: "/test",
        query: { a: 1234 },
        headers: { accept: "application/json" }
      }
    });

    expect(handler(2)).toEqual({ status: 204 });
  });

  test("it should return 417 Expectation failed if times given does not match the number of requests in the requestStore", () => {
    requestStore.push({
      request: {
        method: "GET",
        pathname: "/test",
        query: { a: 1234 },
        headers: { accept: "application/json" }
      }
    });
    requestStore.push({
      request: {
        method: "GET",
        pathname: "/test",
        query: { a: 1234 },
        headers: { accept: "application/json" }
      }
    });

    expect(handler(2)).toEqual(
      expect.objectContaining({
        status: 417
      })
    );
  });

  test("the error message should match the snapshot", () => {
    requestStore.push({
      request: {
        method: "GET",
        pathname: "/test",
        query: { a: 1234 },
        headers: { accept: "application/json" }
      }
    });
    requestStore.push({
      request: {
        method: "POST",
        pathname: "/test/1234",
        body: { a: 1234 },
        headers: { accept: "application/json" }
      }
    });
    requestStore.push({
      request: {
        method: "POST",
        pathname: "/test/abc",
        body: "test",
        headers: { accept: "application/json" }
      }
    });

    expect(handler(3).body).toMatchSnapshot();
  });
});
