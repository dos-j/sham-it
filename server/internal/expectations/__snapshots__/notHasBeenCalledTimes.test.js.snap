// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit: notHasBeenCalledTimes the error message should match the snapshot 1`] = `
"Expected sham not to have been called 3 times, but it was called 3 times:
  GET /test?a=1234
    accept: application/json,
  POST /test/1234
    accept: application/json
    body:
        {\\"a\\":1234},
  POST /test/abc
    accept: application/json
    body:
        test"
`;
